lms/djangoapps/grades/subsection_grade.py

```diff
@@ -107,11 +107,11 @@ def problem_scores(self):
         course.
         """
         locations = OrderedDict()  # dict of problem locations to ProblemScore
-        for block_key in self.course_data.structure.post_order_traversal(
-                filter_func=possibly_scored,
-                start_node=self.location,
+        for block_key in self.course_data.effective_structure.post_order_traversal(
```

Comment:
Am I understanding correctly that, without this change, none of the rest of the changes in this PR matter? We need this lazy property to use the correct `structure`, correct?

Reply:
That's correct - swapping out `structure` for `effective_structure` is the key change, here.